### ISSUE TITLE
Changing mappings has no effect

### ISSUE THREAD

**Issue by D00mch** [NeogitOrg/neogit#1555]
### Description

Can't remap “o“. It always open my github page in the browers no matter what mappings I use. I can see that remappings are applied (UI shows it in the Hint):
![image](https://github.com/user-attachments/assets/dbe5df5b-0f71-4b02-b539-10f515a5a602)

### Neovim version

NVIM v0.10.0
Build type: Release
LuaJIT 2.1.1720049189
Run "nvim -V1 -v" for more info

### Operating system and version

Sonoma 14.4 (23E214)

### Steps to reproduce

1. Make sure mapping for “q“ and “o“ are remapped in config.
2. Open neogit with `:Neogit`
3. Press “q“

### Expected behavior

“O“ behaves as “Toggle“, “q“ do not close.

### Actual behavior

Remappings are ignored AF.

### Minimal config

```Lua
-- NOTE: See the end of this file if you are reporting an issue, etc. Ignore all the "scary" functions up top, those are
-- used for setup and other operations.
local M = {}

local base_root_path = vim.fn.fnamemodify(debug.getinfo(1, "S").source:sub(2), ":p:h") .. "/.min"
function M.root(path)
  return base_root_path .. "/" .. (path or "")
end

function M.load_plugin(plugin_name, plugin_url)
  local package_root = M.root("plugins/")
  local install_destination = package_root .. plugin_name
  vim.opt.runtimepath:append(install_destination)

  if not vim.loop.fs_stat(package_root) then
    vim.fn.mkdir(package_root, "p")
  end

  if not vim.loop.fs_stat(install_destination) then
    print(string.format("> Downloading plugin '%s' to '%s'", plugin_name, install_destination))
    vim.fn.system({
      "git",
      "clone",
      "--depth=1",
      plugin_url,
      install_destination,
    })
    if vim.v.shell_error > 0 then
      error(string.format("> Failed to clone plugin: '%s' in '%s'!", plugin_name, install_destination),
        vim.log.levels.ERROR)
    end
  end
end

---@alias PluginName string The plugin name, will be used as part of the git clone destination
---@alias PluginUrl string The git url at which a plugin is located, can be a path. See https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols for details
---@alias MinPlugins table<PluginName, PluginUrl>

---Do the initial setup. Downloads plugins, ensures the minimal init does not pollute the filesystem by keeping
---everything self contained to the CWD of the minimal init file. Run prior to running tests, reproducing issues, etc.
---@param plugins? table<PluginName, PluginUrl>
function M.setup(plugins)
  vim.opt.packpath = {}                      -- Empty the package path so we use only the plugins specified
  vim.opt.runtimepath:append(M.root(".min")) -- Ensure the runtime detects the root min dir

  -- Install required plugins
  if plugins ~= nil then
    for plugin_name, plugin_url in pairs(plugins) do
      M.load_plugin(plugin_name, plugin_url)
    end
  end

  vim.env.XDG_CONFIG_HOME = M.root("xdg/config")
  vim.env.XDG_DATA_HOME = M.root("xdg/data")
  vim.env.XDG_STATE_HOME = M.root("xdg/state")
  vim.env.XDG_CACHE_HOME = M.root("xdg/cache")

  -- NOTE: Cleanup the xdg cache on exit so new runs of the minimal init doesn't share any previous state, e.g. shada
  vim.api.nvim_create_autocmd("VimLeave", {
    callback = function()
      vim.fn.system({
        "rm",
        "-r",
        "-f",
        M.root("xdg")
      })
    end
  })
end

-- NOTE: If you have additional plugins you need to install to reproduce your issue, include them in the plugins
-- table within the setup call below.
M.setup({
  plenary = "https://github.com/nvim-lua/plenary.nvim.git",
  telescope = "https://github.com/nvim-telescope/telescope.nvim",
  diffview = "https://github.com/sindrets/diffview.nvim",
  neogit = "https://github.com/NeogitOrg/neogit"
})
-- WARN: Do all plugin setup, test runs, reproductions, etc. AFTER calling setup with a list of plugins!
-- Basically, do all that stuff AFTER this line.
require("neogit").setup({mappings = {status = {o = "Toggle", q = false}}}}) -- For instance, setup Neogit
```

**Comment by D00mch** [comment]
Here is the last commit where mappings used to work: 9fb8a932d21e03db2dfcf5137e6bd26f2f927d9f

**Comment by koopa1338** [comment]
I had a quick look into this and it seems like this is some kind of loading/init issue. If postpone the require of the `ProcessBuffer` when it actually is created the keybindings are working again:

```diff
 lua/neogit/process.lua | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/lua/neogit/process.lua b/lua/neogit/process.lua
index afa6ebec..506a7c26 100644
--- a/lua/neogit/process.lua
+++ b/lua/neogit/process.lua
@@ -5,7 +5,7 @@ local config = require("neogit.config")
 local logger = require("neogit.logger")
 local util = require("neogit.lib.util")
 
-local ProcessBuffer = require("neogit.buffers.process")
+local ProcessBuffer = nil
 local Spinner = require("neogit.spinner")
 
 local api = vim.api
@@ -385,6 +385,9 @@ function Process:spawn(cb)
   self.stdin = job
 
   if not hide_console then
+    if not ProcessBuffer then
+      ProcessBuffer = require("neogit.buffers.process")
+    end
     self.buffer = ProcessBuffer:new(self)
     self:show_spinner()
     self:start_timer()
```

**Comment by CKolkey** [comment]
https://github.com/NeogitOrg/neogit/commit/419f08bbdfedcd7a85c7f96b3367eb2a63366ea0 should fix it
